### PR TITLE
FEAT: Finish Login page and add start Register Page

### LINKED ISSUE #39: FEAT: Finish Login page and add start Register Page
- [x] Add css to the login page
- [x] Add the register page
- [x] Add Github Authentication

### PR DESCRIPTION
**Pull request by ips-coding-challenge**
closes #39

### CODE REVIEW

File: src/styles/styles.css [hunk 1]
```diff
@@ -17,6 +17,33 @@
...
+@keyframes lds-dual-ring {
+    0% {
+        transform: rotate(0deg);
+    }
+    100% {
+        transform: rotate(360deg);
+    }
+}
+
 /* width */
```
Comment (fajfara) [inline src/styles/styles.css:47]: Should we follow the same way of applying properties as above/rest of this CSS file (using tailwind and @apply )?

**Review by fajfara — CHANGES_REQUESTED** [review]
The question here is, do we need two separate components/pages for login and register.
As far as I compared, RegisterForm.tsx and LoginForm.tsx are practically the same, the only difference in these two is validation schema-s (which I think can just be extracted for example take email and password store in a separate object, then just spread those properties where needed).
State on both looks the same, the handleLogin and handleRegister methods look the same (only difference is the endpoint), same with handleSubmit minor difference in which handle method is called. The returned jsx also looks the same.
The same goes for LoginPage.tsx and RegisterPage.tsx, I would just refactor into a single page component ie. AuthPage.tsx.

One bug I found, steps I took:
- First, go to /login (no token so it shows the login page)
- Click inside of one of the inputs (Actually this seems weird, I had it sometimes happen when clicking email input, then another password input, most of the time it happens when clicking password input and then clicking register)
- Click on the Register link at the bottom
- It seems like the component won't switch because the form validation fails (empty input)
Also, I can't access /register page? If I go to /register it just redirects me to /login... This might actually be why the above bug happens :S

File: src/styles/styles.css (hunk 1 above)
Comment (ips-coding-challenge) [reply src/styles/styles.css:47]: Actually, I just copy/paste that chunk of code. Maybe you'll want another loading button or just a loader somewhere. Those classes could be completely removed if we do it another way. That's why I let it this way. ;)

**Comment by ips-coding-challenge** [comment]
I am not sure to have only one component instead of two. I know that there are practically identical but I think it's better to have 2 components. Furthermore, I'm not sure if we will not introduce some problems with Formik.

**Comment by fajfara** [comment]
I will try refactoring there shouldn't be any problems with formik. I will work on this branch, but i will make my own locally, if you still want to do some changes on this branch.

**Review by fajfara — APPROVED** [review]

**Comment by fajfara** [comment]
Refactor, also added a simple unstyled logout button when you hover over logo in the navbar

**Pull request merged**